File: User/lib/network/response_models/advertisement_model.dart

class AdvertisementModel {
  String? title;
  String? isPermanent;
  bool? status;
  String? userType;
  String? createdAt;
  String? id;
  String? image;
  bool isAd = false;

  AdvertisementModel(
      {this.title,
        this.isPermanent,
        this.status,
        this.userType,
        this.createdAt,
        this.id,
        this.image});

  AdvertisementModel.fromJson(Map<String, dynamic> json) {
    title = json['title'];
    isPermanent = json['isPermanent'];
    status = json['status'];
    userType = json['userType'];
    createdAt = json['createdAt'];
    id = json['id'];
    image = json['image'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['title'] = this.title;
    data['isPermanent'] = this.isPermanent;
    data['status'] = this.status;
    data['userType'] = this.userType;
    data['createdAt'] = this.createdAt;
    data['id'] = this.id;
    data['image'] = this.image;
    return data;
  }
}

File: Backend/src/models/boilerplate/advertisement.model.js
const mongoose = require('mongoose');
const { toJSON, paginate } = require('../plugins');

const advertisementSchema = mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    isPermanent: {
      type: String,
      enum: ['permanent', 'temporary'],
      required: true,
      default: 'temporary',
    },

    status: {
      type: Boolean,
      required: true,
      default: true,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
      required: true,
    },
    zoneId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Zone',
      required: true,
    },
    userType: {
      type: String,
      enum: ['user', 'driver'],
      required: true,
    },
    image: {
      type: String, // Store image URL or file path
      required: true,
    },
  },
  {
    timestamps: true,
  },
);

// add plugin that converts mongoose to json
advertisementSchema.plugin(toJSON);
advertisementSchema.plugin(paginate);

/**
 * @typedef Advertisement
 */
const Advertisement = mongoose.model('Advertisement', advertisementSchema);

module.exports = Advertisement;
